anchore_engine/services/policy_engine/engine/vulns/mappers.py

```diff
@@ -56,13 +470,27 @@ def transform_image_to_sbom(
         """
         Generate grype sbom from ImagePackage artifacts
         """
-        # initialize sbom
+        distro_mapper_class = ENGINE_DISTRO_MAPPERS.get(image.distro_name)
+        if not distro_mapper_class:
+            log.error(
+                "No distro mapper found for %s. Cannot generate sbom", image.distro_name
+            )
+            raise ValueError(
+                "No distro mapper found for {}. Cannot generate sbom".format(
+                    image.distro_name
+                )
+            )
+        distro_mapper = distro_mapper_class()
```

Comment:
An approach with the lookup map including instances instead of classes makes this simpler and faster too, no need to instantiate over and over unless the instances behave differently.